### PR TITLE
Add test coverage for NumericUpDown

### PR DESCRIPTION
**Pull request by Syareel-Sukeri**
Related https://github.com/dotnet/winforms/issues/10453

## Proposed changes

- Add unit tests for NumericUpDown to test its properties: Hexadecimal, Increment, Maximum, Minimum, ThousandsSeparator, Value
- Add unit tests for NumericUpDown to test its events: PaddingChanged, TextChanged, ValueChanged, 
- Add unit tests for NumericUpDown to test its methods: BeginInit, DownButton, EndInit, ToString, UpButton
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/winforms/pull/11522)

### CODE REVIEW

File: src/System.Windows.Forms/tests/UnitTests/NumericUpDownTests.cs [hunk 1]
```diff
@@ -220,4 +221,626 @@ public void NumericUpDown_VisualStyles_on_BasicRendering()
 
         */
     }
+
+    [WinFormsFact]
+    public void NumericUpDown_Accelerations_Get_ReturnsExpected()
+    {
+        using NumericUpDown upDown = new();
+        NumericUpDownAccelerationCollection accelerations = upDown.Accelerations;
```
Comment (hughbe) [inline src/System.Windows.Forms/tests/UnitTests/NumericUpDownTests.cs:229]: should be same across multiple calls?

File: src/System.Windows.Forms/tests/UnitTests/NumericUpDownTests.cs [hunk 2]
```diff
@@ -220,4 +221,626 @@ public void NumericUpDown_VisualStyles_on_BasicRendering()
...
+        upDown.Minimum = 10;
+        upDown.Maximum.Should().Be(30);
+        upDown.Minimum.Should().Be(10);
+    }
+
+    [WinFormsFact]
+    public void NumericUpDown_Padding_Set_GetReturnsExpected()
+    {
+        using NumericUpDown upDown = new();
+        var padding = new Padding(1, 2, 3, 4);
```
Comment (Zheng-Li01) [inline src/System.Windows.Forms/tests/UnitTests/NumericUpDownTests.cs:365]: Please use target type new:
Padding padding = new(1, 2, 3, 4);

File: src/System.Windows.Forms/tests/UnitTests/NumericUpDownTests.cs [hunk 3]
```diff
@@ -220,4 +221,626 @@ public void NumericUpDown_VisualStyles_on_BasicRendering()
...
+        NumericUpDownAccelerationCollection accelerations = upDown.Accelerations;
+        accelerations.Add(new NumericUpDownAcceleration(2, 1));
+        accelerations.Should().HaveCount(1);
+    }
+
+    [WinFormsFact]
+    public void NumericUpDown_Accelerations_GetNotNullValue_ReturnsExpected()
+    {
+        using NumericUpDown upDown = new();
+        var acceleration = new NumericUpDownAcceleration(2, 1);
```
Comment (Zheng-Li01) [inline src/System.Windows.Forms/tests/UnitTests/NumericUpDownTests.cs:247]: Please use target type new:
NumericUpDownAcceleration acceleration = new(2, 1);

File: src/System.Windows.Forms/tests/UnitTests/NumericUpDownTests.cs [hunk 4]
```diff
@@ -220,4 +221,631 @@ public void NumericUpDown_VisualStyles_on_BasicRendering()
 
         */
     }
+
+    [WinFormsFact]
+    public void NumericUpDown_Accelerations_Get_ReturnsExpected()
+    {
+
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/tests/UnitTests/NumericUpDownTests.cs:228]: Please remove the empty line
